src/main/java/net/cardosi/mojo/TestMojo.java

```diff
@@ -15,301 +15,443 @@
+    } else {
+      failedTests.forEach((name, startupHtmlFile) -> getLog().error(String.format("Test %s failed, please try manually %s",
+                                                                                  name,
+                                                                                  startupHtmlFile)));
+      throw new MojoFailureException("At least one test failed");
     }
-
-    private static boolean isFinished(WebDriver d) {
-        return (Boolean) ((JavascriptExecutor) d).executeScript("return !!(window.G_testRunner && window.G_testRunner.isFinished())");
+  }
+
+  private void analyzeLog(WebDriver driver) {
+    logMessageFromDriver(LogType.BROWSER,
+                         driver);
+    if (driver.getPageSource()
+              .contains("closureTestRunnerLog")) {
+      this.logMessageFromDriverHTML(driver.getPageSource());
     }
-
-    /**
-     * If specific tests are specified, will use them, otherwise will look for the tests through includes/excludes,
-     * and produce a build config each.
-     */
-    public static List<ClosureBuildConfiguration> getTestConfigs(ClosureBuildConfiguration baseConfig, List<String> tests, MavenProject project, List<String> includes, List<String> excludes) {
-        List<String> testEntrypoints;
-        if (tests == null || tests.isEmpty()) {
-            testEntrypoints = project.getTestCompileSourceRoots().stream()
-                    .flatMap(s -> getTestEntrypoints(new File(s), includes, excludes).stream())
-                    .distinct()
-                    .map(f -> f.replaceAll("\\.java$", "").replaceAll("/", "."))
-                    .collect(Collectors.toList());
-        } else {
-            testEntrypoints = tests;
-        }
-        return testEntrypoints.stream().map(name -> new TestConfig(name, baseConfig)).collect(Collectors.toList());
+  }
+
+  private void logMessageFromDriver(String logType,
+                                    WebDriver driver) {
+    if (!Objects.isNull(driver)) {
+      driver.manage()
+            .logs()
+            .get(logType)
+            .getAll()
+            .forEach(l -> {
+              if (Level.SEVERE.equals(l.getLevel())) {
+                getLog().error(String.format(stripText(l.getMessage())));
+              } else {
+                getLog().info(String.format(stripText(l.getMessage())));
+              }
+            });
     }
-
-    private static List<String> getTestEntrypoints(File testSourceDir, List<String> includes, List<String> excludes) {
-        DirectoryScanner scanner = new DirectoryScanner();
-        scanner.setBasedir(testSourceDir);
-        scanner.setIncludes(includes.toArray(new String[0]));
-        if (excludes != null) {
-            scanner.setExcludes(excludes.toArray(new String[0]));
-        }
-        scanner.scan();
-
-        return Arrays.asList(scanner.getIncludedFiles());
+  }
+
+  private void logMessageFromDriverHTML(String pageSource) {
```

Comment:
This method grabs the message printed inside the brwoser window. Not sure, if this message has a benefit, so I printed it out. This method should only be called in case of error.

Reply:
It's the `closureTestRunnerLog`